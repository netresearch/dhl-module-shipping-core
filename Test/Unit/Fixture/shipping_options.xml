<?xml version="1.0"?>
<carriers xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:module:Dhl_ShippingCore/etc/shipping_options.xsd">
    <carrier code="base">
        <metadata>
            <imageUrl>Dhl_Ui::images/logo-dhl-wide.svg</imageUrl>
        </metadata>
    </carrier>
    <carrier code="dhlpaket">
        <serviceOptions>
            <shippingOption code="parcelAnnouncement" available="dhlshippingsolutions/dhlpaket/additional_services/services_group/parcelannouncement">
                <label>Parcel Announcement</label>
                <sortOrder>10</sortOrder>
                <inputs>
                    <input code="enabled">
                        <inputType>checkbox</inputType>
                        <label>Enable DHL parcel notification</label>
                        <tooltip>Your e-mail address will be sent to DHL upon enabling, after which DHL will trigger a package announcement for your shipment. The e-mail address will exclusively be used for the announcement of this shipment.</tooltip>
                    </input>
                </inputs>
            </shippingOption>
            <shippingOption code="parcelstation" available="dhlshippingsolutions/dhlpaket/additional_services/services_group/parcelstation">
                <label>Parcel station delivery: Delivery to a parcel station of your choice</label>
                <sortOrder>20</sortOrder>
                <routes>
                    <route>
                        <origin>DE</origin>
                        <includeDestinations>
                            <destination>DE</destination>
                            <destination>AT</destination>
                        </includeDestinations>
                    </route>
                </routes>
                <inputs>
                    <input code="enabled">
                        <label>Select parcel station</label>
                        <labelVisible>false</labelVisible>
                        <inputType>checkbox</inputType>
                    </input>
                    <input code="id">
                        <label>Parcel Station ID</label>
                        <labelVisible>true</labelVisible>
                        <inputType>select</inputType>
                        <placeholder>Please select a parcel station &#x2026;</placeholder>
                        <comment>
                            <content>Select a nearby parcel shop. Your package will be sent there for you to pick up.</content>
                        </comment>
                        <validationRules><rule name="required-entry"/></validationRules>
                    </input>
                    <input code="postNumber">
                        <label>Post number</label>
                        <labelVisible>true</labelVisible>
                        <inputType>text</inputType>
                        <placeholder>Post number</placeholder>
                        <validationRules>
                            <rule name="maxLength">
                                <param>10</param>
                            </rule>
                            <rule name="validate-no-html-tags"/>
                            <rule name="dhl_filter_special_chars"/>
                            <rule name="required-entry"/>
                        </validationRules>
                    </input>
                </inputs>
            </shippingOption>
            <shippingOption code="preferredDay" available="dhlshippingsolutions/dhlpaket/additional_services/services_group/preferredday">
                <label>Preferred Day: Delivery on your preferred day</label>
                <sortOrder>30</sortOrder>
                <routes>
                    <route>
                        <includeDestinations>
                            <destination>DE</destination>
                            <destination>AT</destination>
                        </includeDestinations>
                    </route>
                </routes>
                <inputs>
                    <input code="date">
                        <label>Preferred Day</label>
                        <labelVisible>false</labelVisible>
                        <inputType>date</inputType>
                        <comment>
                            <content><![CDATA[This service will incur additional shipping costs of $1 including VAT.]]></content>
                            <footnoteId>footnote-combined-cost</footnoteId>
                        </comment>
                        <tooltip>Choose one of the displayed days as your preferred day for your parcel delivery. Other days are not possible due to delivery processes.</tooltip>
                        <options>
                            <option id="none">
                                <label>none</label>
                            </option>
                        </options>
                    </input>
                </inputs>
            </shippingOption>
            <shippingOption code="preferredTime" available="dhlshippingsolutions/dhlpaket/additional_services/services_group/preferredtime">
                <label>Preferred Time: Delivery at your preferred time</label>
                <sortOrder>40</sortOrder>
                <inputs>
                    <input code="time">
                        <inputType>time</inputType>
                        <label>Enable Preferred Time</label>
                        <labelVisible>false</labelVisible>
                        <comment>
                            <content><![CDATA[This service will incur additional shipping costs of $1 including VAT.]]></content>
                            <footnoteId>footnote-combined-cost</footnoteId>
                        </comment>
                        <tooltip>Indicate a preferred time, which suits you best for your parcel delivery by choosing one of the displayed time windows.</tooltip>
                        <options>
                            <option id="none">
                                <label>none</label>
                            </option>
                        </options>
                    </input>
                </inputs>
            </shippingOption>
            <shippingOption code="preferredLocation" available="dhlshippingsolutions/dhlpaket/additional_services/services_group/preferredlocation">
                <label>Preferred Location: Delivery to your preferred drop-off location</label>
                <sortOrder>50</sortOrder>
                <routes>
                    <route>
                        <includeDestinations>
                            <destination>
                                DE
                            </destination>
                            <destination>AT</destination>
                        </includeDestinations>
                    </route>
                </routes>
                <inputs>
                    <input code="details">
                        <sortOrder>20</sortOrder>
                        <label>Preferred Location</label>
                        <labelVisible>false</labelVisible>
                        <inputType>text</inputType>
                        <tooltip>Choose a weather-protected and non-visible place on your property where we can deposit the parcel in your absence.</tooltip>
                        <placeholder>E.g. garage, terrace</placeholder>
                        <validationRules>
                            <rule name="maxLength">
                                <param>40</param>
                            </rule>
                            <rule name="validate-no-html-tags"/>
                            <rule name="dhl_filter_special_chars"/>
                            <rule name="dhl_filter_packing_station"/>
                        </validationRules>
                    </input>
                </inputs>
            </shippingOption>
            <shippingOption code="preferredNeighbour" available="dhlshippingsolutions/dhlpaket/additional_services/services_group/preferredneighbour">
                <label>Preferred Neighbour: Delivery to a neighbour of your choice</label>
                <sortOrder>60</sortOrder>
                <routes>
                    <route>
                        <includeDestinations>
                            <destination>DE</destination>
                            <destination>AT</destination>
                        </includeDestinations>
                    </route>
                </routes>
                <inputs>
                    <input code="name">
                        <sortOrder>20</sortOrder>
                        <label>Neighbour name</label>
                        <inputType>text</inputType>
                        <tooltip>Declare a person in your immediate neighborhood whom we can hand out your parcel. This person should live in the same building, directly opposite or next door.</tooltip>
                        <placeholder>First name, last name of neighbour</placeholder>
                        <validationRules>
                            <rule name="maxLength">
                                <param>40</param>
                            </rule>
                            <rule name="validate-no-html-tags"/>
                            <rule name="dhl_filter_special_chars"/>
                            <rule name="dhl_filter_packing_station"/>
                        </validationRules>
                    </input>
                    <input code="address">
                        <sortOrder>30</sortOrder>
                        <label>Neighbour address</label>
                        <inputType>text</inputType>
                        <placeholder>Street, number, postal code, city</placeholder>
                        <validationRules>
                            <rule name="maxLength">
                                <param>40</param>
                            </rule>
                            <rule name="validate-no-html-tags"/>
                            <rule name="dhl_filter_special_chars"/>
                        </validationRules>
                    </input>
                </inputs>
            </shippingOption>
        </serviceOptions>
        <metadata>
            <title>DHL Preferred Delivery. Delivered just the way you want.</title>
            <commentsBefore>
                <comment>
                    <content>You decide when and where you want to receive your parcels with the recipient services of DHL.</content>
                </comment>
                <comment>
                    <content>Please choose your preferred delivery options.</content>
                </comment>
            </commentsBefore>
            <footnotes>
                <footnote id="footnote-combined-cost">
                    <content><![CDATA[For a combined booking of preferred day and preferred time, the shipping costs are a total of $1 including VAT.]]></content>
                    <subjects>
                        <subject>preferredTime</subject>
                        <subject>preferredDay</subject>
                    </subjects>
                    <subjectsMustBeAvailable>true</subjectsMustBeAvailable>
                    <subjectsMustBeSelected>true</subjectsMustBeSelected>
                </footnote>
            </footnotes>
        </metadata>
        <compatibilityData>
            <rule id="preferredLocationAndNeighbourExclusion">
                <triggerValue>*</triggerValue>
                <action>disable</action>
                <masters>
                    <master>preferredLocation</master>
                </masters>
                <subjects>
                    <subject>preferredNeighbour</subject>
                </subjects>
                <errorMessage>Please choose only one of %1.</errorMessage>
            </rule>
            <rule id="preferredLocationAndNeighbourExclusion2">
                <triggerValue>*</triggerValue>
                <action>disable</action>
                <masters>
                    <master>preferredNeighbour</master>
                </masters>
                <subjects>
                    <subject>preferredLocation</subject>
                </subjects>
                <errorMessage>Please choose only one of %1.</errorMessage>
            </rule>
            <rule id="preferredNeighbourRequireChildren">
                <triggerValue>*</triggerValue>
                <action>require</action>
                <subjects>
                    <subject>preferredNeighbour.name</subject>
                    <subject>preferredNeighbour.address</subject>
                </subjects>
                <errorMessage>Some values for Preferred Neighbour service are missing.</errorMessage>
            </rule>
            <rule id="parcelstationShowMore">
                <triggerValue>true</triggerValue>
                <action>show</action>
                <masters>
                    <master>parcelstation.enabled</master>
                </masters>
                <subjects>
                    <subject>parcelstation.id</subject>
                    <subject>parcelstation.postNumber</subject>
                </subjects>
            </rule>
            <rule id="parcelstationRequireChildren">
                <triggerValue>true</triggerValue>
                <action>require</action>
                <masters>
                    <master>parcelstation.enabled</master>
                </masters>
                <subjects>
                    <subject>parcelstation.id</subject>
                    <subject>parcelstation.postNumber</subject>
                </subjects>
            </rule>
            <rule id="parcelstationExcludeAllOthers">
                <masters>
                    <master>parcelstation.enabled</master>
                </masters>
                <subjects>
                    <subject>preferredDay</subject>
                    <subject>preferredTime</subject>
                    <subject>preferredLocation</subject>
                    <subject>preferredNeighbour</subject>
                    <subject>parcelOutletRouting</subject>
                </subjects>
                <triggerValue>true</triggerValue>
                <action>hide</action>
            </rule>
            <rule id="allOthersExcludeParcelstation">
                <masters>
                    <master>preferredDay</master>
                    <master>preferredTime</master>
                    <master>preferredLocation</master>
                    <master>preferredNeighbour</master>
                    <master>parcelOutletRouting</master>
                </masters>
                <subjects>
                    <subject>parcelstation.enabled</subject>
                </subjects>
                <triggerValue>*</triggerValue>
                <action>hide</action>
            </rule>
        </compatibilityData>
    </carrier>
</carriers>
