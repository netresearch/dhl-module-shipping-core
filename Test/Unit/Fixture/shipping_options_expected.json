{
  "carriers": {
    "base": {
      "metadata": {
        "imageUrl": "Dhl_Ui::images\/logo-dhl-wide.svg"
      },
      "code": "base"
    },
    "dhlpaket": {
      "serviceOptions": {
        "parcelAnnouncement": {
          "label": "Parcel Announcement",
          "sortOrder": 10,
          "inputs": {
            "enabled": {
              "inputType": "checkbox",
              "label": "Enable DHL parcel notification",
              "tooltip": "Your e-mail address will be sent to DHL upon enabling, after which DHL will trigger a package announcement for your shipment. The e-mail address will exclusively be used for the announcement of this shipment.",
              "code": "enabled"
            }
          },
          "code": "parcelAnnouncement",
          "available": "dhlshippingsolutions\/dhlpaket\/additional_services\/services_group\/parcelannouncement"
        },
        "parcelstation": {
          "label": "Parcel station delivery: Delivery to a parcel station of your choice",
          "sortOrder": 20,
          "routes": [
            {
              "origin": "DE",
              "includeDestinations": [
                "DE",
                "AT"
              ]
            }
          ],
          "inputs": {
            "enabled": {
              "label": "Select parcel station",
              "labelVisible": false,
              "inputType": "checkbox",
              "code": "enabled"
            },
            "id": {
              "label": "Parcel Station ID",
              "labelVisible": true,
              "inputType": "select",
              "placeholder": "Please select a parcel station \u2026",
              "comment": {
                "content": "Select a nearby parcel shop. Your package will be sent there for you to pick up."
              },
              "validationRules": {
                "required-entry": {
                  "name": "required-entry"
                }
              },
              "code": "id"
            },
            "postNumber": {
              "label": "Post number",
              "labelVisible": true,
              "inputType": "text",
              "placeholder": "Post number",
              "validationRules": {
                "maxLength": {
                  "param": 10,
                  "name": "maxLength"
                },
                "validate-no-html-tags": {
                  "name": "validate-no-html-tags"
                },
                "dhl_filter_special_chars": {
                  "name": "dhl_filter_special_chars"
                },
                "required-entry": {
                  "name": "required-entry"
                }
              },
              "code": "postNumber"
            }
          },
          "code": "parcelstation",
          "available": "dhlshippingsolutions\/dhlpaket\/additional_services\/services_group\/parcelstation"
        },
        "preferredDay": {
          "label": "Preferred Day: Delivery on your preferred day",
          "sortOrder": 30,
          "routes": [
            {
              "includeDestinations": [
                "DE",
                "AT"
              ]
            }
          ],
          "inputs": {
            "date": {
              "label": "Preferred Day",
              "labelVisible": false,
              "inputType": "date",
              "comment": {
                "content": "This service will incur additional shipping costs of $1 including VAT.",
                "footnoteId": "footnote-combined-cost"
              },
              "tooltip": "Choose one of the displayed days as your preferred day for your parcel delivery. Other days are not possible due to delivery processes.",
              "options": {
                "none": {
                  "label": "none",
                  "id": "none"
                }
              },
              "code": "date"
            }
          },
          "code": "preferredDay",
          "available": "dhlshippingsolutions\/dhlpaket\/additional_services\/services_group\/preferredday"
        },
        "preferredTime": {
          "label": "Preferred Time: Delivery at your preferred time",
          "sortOrder": 40,
          "inputs": {
            "time": {
              "inputType": "time",
              "label": "Enable Preferred Time",
              "labelVisible": false,
              "comment": {
                "content": "This service will incur additional shipping costs of $1 including VAT.",
                "footnoteId": "footnote-combined-cost"
              },
              "tooltip": "Indicate a preferred time, which suits you best for your parcel delivery by choosing one of the displayed time windows.",
              "options": {
                "none": {
                  "label": "none",
                  "id": "none"
                }
              },
              "code": "time"
            }
          },
          "code": "preferredTime",
          "available": "dhlshippingsolutions\/dhlpaket\/additional_services\/services_group\/preferredtime"
        },
        "preferredLocation": {
          "label": "Preferred Location: Delivery to your preferred drop-off location",
          "sortOrder": 50,
          "routes": [
            {
              "includeDestinations": [
                "DE",
                "AT"
              ]
            }
          ],
          "inputs": {
            "details": {
              "sortOrder": 20,
              "label": "Preferred Location",
              "labelVisible": false,
              "inputType": "text",
              "tooltip": "Choose a weather-protected and non-visible place on your property where we can deposit the parcel in your absence.",
              "placeholder": "E.g. garage, terrace",
              "validationRules": {
                "maxLength": {
                  "param": 40,
                  "name": "maxLength"
                },
                "validate-no-html-tags": {
                  "name": "validate-no-html-tags"
                },
                "dhl_filter_special_chars": {
                  "name": "dhl_filter_special_chars"
                },
                "dhl_filter_packing_station": {
                  "name": "dhl_filter_packing_station"
                }
              },
              "code": "details"
            }
          },
          "code": "preferredLocation",
          "available": "dhlshippingsolutions\/dhlpaket\/additional_services\/services_group\/preferredlocation"
        },
        "preferredNeighbour": {
          "label": "Preferred Neighbour: Delivery to a neighbour of your choice",
          "sortOrder": 60,
          "routes": [
            {
              "includeDestinations": [
                "DE",
                "AT"
              ]
            }
          ],
          "inputs": {
            "name": {
              "sortOrder": 20,
              "label": "Neighbour name",
              "inputType": "text",
              "tooltip": "Declare a person in your immediate neighborhood whom we can hand out your parcel. This person should live in the same building, directly opposite or next door.",
              "placeholder": "First name, last name of neighbour",
              "validationRules": {
                "maxLength": {
                  "param": 40,
                  "name": "maxLength"
                },
                "validate-no-html-tags": {
                  "name": "validate-no-html-tags"
                },
                "dhl_filter_special_chars": {
                  "name": "dhl_filter_special_chars"
                },
                "dhl_filter_packing_station": {
                  "name": "dhl_filter_packing_station"
                }
              },
              "code": "name"
            },
            "address": {
              "sortOrder": 30,
              "label": "Neighbour address",
              "inputType": "text",
              "placeholder": "Street, number, postal code, city",
              "validationRules": {
                "maxLength": {
                  "param": 40,
                  "name": "maxLength"
                },
                "validate-no-html-tags": {
                  "name": "validate-no-html-tags"
                },
                "dhl_filter_special_chars": {
                  "name": "dhl_filter_special_chars"
                }
              },
              "code": "address"
            }
          },
          "code": "preferredNeighbour",
          "available": "dhlshippingsolutions\/dhlpaket\/additional_services\/services_group\/preferredneighbour"
        }
      },
      "metadata": {
        "title": "DHL Preferred Delivery. Delivered just the way you want.",
        "commentsBefore": [
          {
            "content": "You decide when and where you want to receive your parcels with the recipient services of DHL."
          },
          {
            "content": "Please choose your preferred delivery options."
          }
        ],
        "footnotes": {
          "footnote-combined-cost": {
            "content": "For a combined booking of preferred day and preferred time, the shipping costs are a total of $1 including VAT.",
            "subjects": [
              "preferredTime",
              "preferredDay"
            ],
            "subjectsMustBeAvailable": true,
            "subjectsMustBeSelected": true,
            "id": "footnote-combined-cost"
          }
        }
      },
      "compatibilityData": {
        "preferredLocationAndNeighbourExclusion": {
          "triggerValue": "*",
          "action": "disable",
          "masters": [
            "preferredLocation"
          ],
          "subjects": [
            "preferredNeighbour"
          ],
          "errorMessage": "Please choose only one of %1.",
          "id": "preferredLocationAndNeighbourExclusion"
        },
        "preferredLocationAndNeighbourExclusion2": {
          "triggerValue": "*",
          "action": "disable",
          "masters": [
            "preferredNeighbour"
          ],
          "subjects": [
            "preferredLocation"
          ],
          "errorMessage": "Please choose only one of %1.",
          "id": "preferredLocationAndNeighbourExclusion2"
        },
        "preferredNeighbourRequireChildren": {
          "triggerValue": "*",
          "action": "require",
          "subjects": [
            "preferredNeighbour.name",
            "preferredNeighbour.address"
          ],
          "errorMessage": "Some values for Preferred Neighbour service are missing.",
          "id": "preferredNeighbourRequireChildren"
        },
        "parcelstationShowMore": {
          "triggerValue": true,
          "action": "show",
          "masters": [
            "parcelstation.enabled"
          ],
          "subjects": [
            "parcelstation.id",
            "parcelstation.postNumber"
          ],
          "id": "parcelstationShowMore"
        },
        "parcelstationRequireChildren": {
          "triggerValue": true,
          "action": "require",
          "masters": [
            "parcelstation.enabled"
          ],
          "subjects": [
            "parcelstation.id",
            "parcelstation.postNumber"
          ],
          "id": "parcelstationRequireChildren"
        },
        "parcelstationExcludeAllOthers": {
          "masters": [
            "parcelstation.enabled"
          ],
          "subjects": [
            "preferredDay",
            "preferredTime",
            "preferredLocation",
            "preferredNeighbour",
            "parcelOutletRouting"
          ],
          "triggerValue": true,
          "action": "hide",
          "id": "parcelstationExcludeAllOthers"
        },
        "allOthersExcludeParcelstation": {
          "masters": [
            "preferredDay",
            "preferredTime",
            "preferredLocation",
            "preferredNeighbour",
            "parcelOutletRouting"
          ],
          "subjects": [
            "parcelstation.enabled"
          ],
          "triggerValue": "*",
          "action": "hide",
          "id": "allOthersExcludeParcelstation"
        }
      },
      "code": "dhlpaket"
    }
  }
}
